drivers/scheduler/k8s/k8s.go

```diff
@@ -6676,16 +6672,88 @@ func (k *K8s) addAnnotationsToPVC(pvc *corev1.PersistentVolumeClaim, annotations
 
 // GetAutopilotNamespace returns the autopilot namespace
 func (k *K8s) GetAutopilotNamespace() (string, error) {
+	var pods *corev1.PodList
+	var err error
+	var ns string
+	ns, err = k.GetPortworxNamespace()
+	if err != nil {
+		if strings.Contains(err.Error(), PortworxNotDeployed) {
+			//Return default AutopilotNamespace
+			return autopilotDefaultNamespace, nil
+		}
+		return "", fmt.Errorf("Failed to find portworx namespace %v", err)
+	}
+	log.Infof("Portworx installed on namespace %s", ns)
+	//Find out autopilot pod deployed namespace by label
+	pods, err = k8sCore.ListPods(map[string]string{
+		AutopilotLabelNameKey: AutopilotLabelValue,
+	})
+	if err != nil {
+		return ns, err
+	}
+	ns = ""
+	if len(pods.Items) > 0 {
+		ns = pods.Items[0].Namespace
+	}
+	if len(ns) == 0 {
+		return ns, fmt.Errorf("Failed to find autopilot namespace using pods with label [%s=%s]", AutopilotLabelNameKey, AutopilotLabelValue)
+	}
+	return ns, nil
+}
+
+// IsAutopilotEnabled returns the autopilot enabled or not
+func (k *K8s) IsAutopilotEnabled() (bool, error) {
+	var pods *corev1.PodList
+	var err error
+	//Find out autopilot pod deployed or not
+	pods, err = k8sCore.ListPods(map[string]string{
+		AutopilotLabelNameKey: AutopilotLabelValue,
+	})
+	if err != nil {
+		return false, err
+	}
+	if len(pods.Items) == 0 {
+		return false, fmt.Errorf("Failed to find autopilot pods with label [%s=%s]", AutopilotLabelNameKey, AutopilotLabelValue)
+	}
+	return true, nil
+}
+
+// GetPortworxNamespace returns namespace where Portworx is deployed based on the portworx-service location
+func (k *K8s) GetPortworxNamespace() (string, error) {
 	allServices, err := k8sCore.ListServices("", metav1.ListOptions{})
 	if err != nil {
 		return "", err
 	}
+
+	var namespaces []string
 	for _, svc := range allServices.Items {
 		if svc.Name == portworxServiceName {
-			return svc.Namespace, nil
+			namespaces = append(namespaces, svc.Namespace)
+		}
+	}
+	if len(namespaces) > 0 {
+		// If portworx-service is deployed in 2 namespaces, return the none kube-system namespace
+		if len(namespaces) == 2 {
+			log.Debugf("Found [%s] service in 2 different namespaces %s", portworxServiceName, namespaces)
+			for _, namespace := range namespaces {
+				if namespace != "kube-system" {
+					log.Debugf("When Portworx deployed outside of [kube-system] namespace, it also creates [%s] service in the [kube-system ] namespace as well as in the namespace it is deployed in", portworxServiceName)
```

Comment:
nit: extra space `[kube-system ]`
```suggestion
					log.Debugf("When Portworx deployed outside of [kube-system] namespace, it also creates [%s] service in the [kube-system] namespace as well as in the namespace it is deployed in", portworxServiceName)
```